pkg/agent/controller/egress/egress_controller.go

```diff
@@ -223,6 +267,19 @@ func (c *EgressController) Run(stopCh <-chan struct{}) {
 		return
 	}
 
+	go c.cluster.Run(stopCh)
+
+	// The Egress has been deleted but assigned IP has not been deleted,
+	// agent should delete those IPs when it starts.
+	for _, ip := range c.ipAssigner.AssignedIPs() {
+		egresses, _ := c.egressInformer.GetIndexer().ByIndex(egressIPIndex, ip)
```

Comment:
I think it should just check if the original egress is still there. If it's deleted, unassign the IP unconditionally because if the IP is assigned to another Egress, the following processing of existing Egresses will process it anyway. If it skips unassigning because it is assigned to another Egress, no worker for existing Egresses will unassign the IP.